### ISSUE TITLE
Cannot pull value from anserValueSet

### ISSUE THREAD

**Issue by yunwwang** [lhncbc/lforms#30]
I create a questionnaire with contained value set:
```
{
  "resourceType": "Questionnaire",
  "id": "HomeOxygenTherapy",
  "name": "HomeOxygenTherapy",
  "meta": {
    "profile": [
      "http://hl7.org/fhir/StructureDefinition/cqif-questionnaire",
      "http://hl7.org/fhir/us/davinci-dtr/StructureDefinition/dtr-questionnaire-r4"
    ]
  },
  "title": "Home Oxygen Therapy",
  "status": "draft",
  "subjectType": [
    "Patient"
  ],
  "contained": [
    {
      "resourceType": "ValueSet",
      "id": "gender",
      "name": "Gender",
      "status": "draft",
      "expansion": {
        "contains": [
          {
            "code": "male",
            "system": "http://hl7.org/fhir/administrative-gender",
            "display": "Male"
          },
          {
            "code": "female",
            "system": "http://hl7.org/fhir/administrative-gender",
            "display": "Female"
          },
          {
            "code": "other",
            "system": "http://hl7.org/fhir/administrative-gender",
            "display": "Other"
          }
        ]
      }
    }
  ],
  "item": [
    {
      "linkId": "1",
      "text": "Patient Information",
      "type": "group",
      "item": [
        {
          "linkId": "1.5",
          "text": "Gender",
          "type": "choice",
          "required": true,
          "answerValueSet": "#gender"
        }
      ]
    }
  ]
}
```
I tested this on https://lhcforms.nlm.nih.gov/lhcforms. The form does NOT show the list of options from the contained value set. I tried to change answerValueSet to `http://hl7.org/fhir/ValueSet/administrative-gender`. The result is the same.

**Comment by plynchnlm** [comment]
See https://github.com/lhncbc/lforms/blob/master/test/data/R4/argonaut-phq9-ish.json for an example of a form that uses answerValueSet to reference a contained ValueSet.  I think the field you are missing in your ValueSets is "url".

**Comment by yunwwang** [comment]
Thanks for the example. My understanding from the sample code is that the anserValueSet element uses the contained value set's url as its value. This is not correct. According to FHIR standard, the canonical url for contained resource is '#' + resource.id

**Comment by plynchnlm** [comment]
It looks like you are correct, although the [spec](http://hl7.org/fhir/references.html#canonical) says, confusingly, "References to canonical URLs use the canonical type to refer to the url element".

**Comment by plynchnlm** [comment]
This was fixed in 24.1.2.
